### ISSUE TITLE
Linux Application.Top renders only after scaling terminal window

### ISSUE THREAD

**Issue by JesFo** [gui-cs/Terminal.Gui#2715]
**Description**
I'm trying to run my application on Debian, but my `Application.Top` renders only after i resize my terminal window. This works just fine on Windows.
I can write to the terminal with `System.Console` just fine on Debian before attempting to call `Application.Run()`.
But unlike on Windows, where the `Application.Top` is rendered immediately after `Application.Run()` is called, The terminal window stays blank on Debian until the terminal window is resized.

**Program entrypoint**
```
    static void Main(string[] args)
    {
        Application.QuitKey = Key.Esc;
        Application.Init();
        _iWnd = new AppWindow("SubTopic",_padding);
        Application.Top.Resized += Top_Resized;
        Application.Top.Add(_iWnd._top);
        Application.Run(Application.Top);
        Application.Shutdown();
    }

```
The class `AppWindow` is called just to add padding to the window drawn using a struct.
AppWindow class constructor:
```
        public AppWindow(string title,Padding _pad) {
            Colors.Base.Normal = Application.Driver.MakeAttribute(Color.White, Color.Black);
            _padding = _pad;

            _top = new Toplevel()
            {
                X = 0,
                Y = 0,
                Width = Dim.Fill(),
                Height = Dim.Fill(),
            };

            _window = new Window()
            {
                LayoutStyle = LayoutStyle.Absolute,
                CanFocus = false,
                ColorScheme = Colors.Base,
                Title = title,
                X = _pad.Left,
                Y = _pad.Top,
                Width = Dim.Fill(_pad.Right),
                Height = Dim.Fill(_pad.Bottom),
            };
            //Resize();
            _top.Add(_window);
            _window.Add(_labels);
            Resize();
        }
```
And the class also includes a function to add labels to the window according to the display size when initialized:
```
        public void Resize()
        {
            int winHeight;
            _window.GetCurrentHeight(out winHeight);
            if (winHeight == 0) return;
            _labels = new Label[winHeight-1];
            Debug.WriteLine($"  Labels:{_labels.Length}");
            for (int y = 1; y < _labels.Length; y += 4)
            {
                _labels[y] = new Label("┤"+y) { ColorScheme = Colors.Base, X = _padding.Left, Y = y + _padding.Top, };
            }
            _top.Clear();
            _top.Add(_labels);
        }
```
The `Resize` mehod is called whenever `Application.Top.Resized` is invoked.

The same issue persists even with a minimal Window application setup.

**Build info**
Target Framework:      .NET7.0
Deployment mode:    Self-contained
Target Runtime:          Linux-x86

**Target OS/Shell info**
OS:                  Debian GNU/Linux 11 (WSL2)
Codename:     Bullseye
Shell:               Tested on : Bash, ZSH, TMUX

**Comment by tznind** [comment]
Hi thanks for providing such a detailed summary of your issue.

You only redraw on Resize so I think the issue is that linux is not giving you a Resized event while windows driver is.

You should be able to perform the initial layout by using the `Loaded` event.

One thing to note is that modifying a views contents during loaded will cause an error unless you queue the action with `Invoke`, see below:

```csharp
_window.Loaded += ()=>Application.MainLoop.Invoke(Resize);
```

**Comment by JesFo** [comment]
> Hi thanks for providing such a detailed summary of your issue.
> 
> You only redraw on Resize so I think the issue is that linux is not giving you a Resized event (during load) while windows driver is.
> 
> You should be able to perform the initial layout by using the `Loaded` event.
> 
> One thing to note is that modifying a views contents during loaded will cause an error unless you queue the action with `Invoke`, see below:
> 
> ```cs
> _window.Loaded += ()=>Application.MainLoop.Invoke(Resize);
> ```
Thank you.
This method works on all shells but TMUX.

Should this be considered a bug, since the behaviour is inconsistent between platforms with identical code?

**Comment by tznind** [comment]
> Should this be considered a bug, since the behaviour is inconsistent between platforms with identical code?

I would say this is more down to the terminal than our API (so not a bug).  Resizing is an event spawned by the terminal itself (i.e. Terminal.Gui only forwards it on).  If a given terminal raises a size change event before/during loading but another one doesn't thats fine.  In fact `Resized` is renamed to `TerminalResized` in version 2 of Terminal.Gui.

The `Loaded` event is more correct place for initialization since its called by Terminal.Gui when the window is presented.

> This method works on all shells but TMUX.

I'm interested in the behaviour of TMUX and how it interacts with Terminal.Gui in general.  Terminal.Gui works pretty well over remote sessions (SSH, Guacamole, Putty etc) but I'm always interested in seeing how it interacts with environments. 

I would guess that the `Loaded` event is still resolving but that maybe something is clearing the screen or otherwise breaking things.  If you could do a more indepth test of TMUX with UICatalog to find out what works and what doesn't that would be great.

**Comment by tig** [comment]
Closing as won't fix: Use Loaded instead.

**Comment by BDisp** [comment]
@JesFo , I was playing with this and the problem isn't TMUX, but with some others Linux terminal. You have to use both events (Resize and Loaded). First of all I have to write this command (`export TERM=xterm-256color`) on TMUX to have the colors. With the code bellow I could have it always working on load and resizing:

```cs
		static void Main(string[] args)
		{
			Application.QuitKey = Key.Esc;
			Application.Init ();
			_iWnd = new AppWindow ("SubTopic", _padding);
			Application.Top.Resized += Top_Resized;
			Application.Top.Loaded += Top_Loaded;
			Application.Top.Add (_iWnd._top);
			Application.Run (Application.Top);
			_iWnd.Dispose ();
			Application.Shutdown ();
		}
		
		private void Top_Loaded ()
		{
			_iWnd.Resize ();
		}

		private void Top_Resized (Size size)
		{
			_iWnd.Resize ();
		}

		public struct Padding {
			public int Left;
			public int Right;
			public int Top;
			public int Bottom;

			public Padding (int len)
			{
				Left = len;
				Right = len;
				Top = len;
				Bottom = len;
			}
		}

		public class AppWindow : Window {
			public Toplevel _top;
			private Window _window;
			private Label [] _labels;

			public AppWindow (string title, Padding _pad)
			{
				Colors.Base.Normal = Application.Driver.MakeAttribute (Color.White, Color.Black);
				_padding = _pad;

				_top = new Toplevel () {
					X = 0,
					Y = 0,
					Width = Dim.Fill (),
					Height = Dim.Fill (),
				};

				_window = new Window () {
					LayoutStyle = LayoutStyle.Absolute,
					CanFocus = false,
					ColorScheme = Colors.Base,
					Title = title,
					X = _pad.Left,
					Y = _pad.Top,
					Width = Dim.Fill (_pad.Right),
					Height = Dim.Fill (_pad.Bottom),
				};
				_top.Add (_window);
				Resize ();
			}

			public void Resize ()
			{
				int winHeight;
				_window.GetCurrentHeight (out winHeight);
				if (_labels != null) {
					foreach (var view in _labels) {
						_top.Remove (view);
						view?.Dispose ();
					}
				}
				if (winHeight == 0) return;
				_labels = new Label [(winHeight / 4) + (winHeight % 4)];
				Debug.WriteLine ($"  Labels:{_labels.Length}");
				int i = 0;
				for (int y = 1; y < winHeight; y += 4) {
					_labels [i] = new Label ("┤" + y) { ColorScheme = Colors.Base, X = _padding.Left, Y = y + _padding.Top, };
					if (y + 4 < winHeight) {
						i++;
					}
				}
				_top.Clear ();
				_top.Add (_labels);
			}
		}

```

![tmux](https://github.com/gui-cs/Terminal.Gui/assets/13117724/a2a66953-197c-41b6-908c-245b00b05828)
